test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/CompareAttributeAdapterTest.cs

```diff
@@ -62,15 +74,25 @@ public void ClientRulesWithCompareAttribute_ErrorMessageUsesDisplayName()
             var adapter = new CompareAttributeAdapter(attribute, stringLocalizer: stringLocalizer.Object);
 
             var actionContext = new ActionContext();
-            var context = new ClientModelValidationContext(actionContext, metadata, metadataProvider);
+            var context = new ClientModelValidationContext(
+                actionContext,
+                metadata,
+                metadataProvider,
+                new AttributeDictionary());
 
             // Act
-            var rules = adapter.GetClientValidationRules(context);
+            adapter.AddValidation(context);
 
             // Assert
-            var rule = Assert.Single(rules);
-            // Mono issue - https://github.com/aspnet/External/issues/19
```

Comment:
Please copy this above the line that sets `expectedMessage`. That will make it much easier to clean up when / if we move to a version of Mono containing a fix. Same for other places we're currently losing this information.